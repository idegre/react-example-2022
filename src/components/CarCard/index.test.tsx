import React from 'react';
import { CarCard } from '.';
import renderer from 'react-test-renderer';

describe('test selectors', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<CarCard car={{
				id: 1,
				type: 'type1',
				brand: 'brand1',
				colors: ['1', '2'],
				img: 'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg/520px-Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg'
			}}/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('handles broken image links', () => {
		const tree = renderer
			.create(<CarCard car={{
				id: 3,
				type: 'type2',
				brand: 'brand',
				colors: ['1', '2', '5'],
				img: 'fake URL'
			}}/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});