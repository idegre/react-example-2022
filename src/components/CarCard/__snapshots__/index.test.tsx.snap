// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test selectors handles broken image links 1`] = `
<div
  className="sc-gsnTZi fnwWxp"
>
  <img
    onError={[Function]}
    src="fake URL"
  />
  <div
    className="sc-dkzDqf cWPqBb"
  >
    <div
      className="sc-bczRLJ hJhdoW"
    >
      brand
    </div>
    <div
      className="sc-bczRLJ hJhdoW"
    >
      type2
    </div>
  </div>
</div>
`;

exports[`test selectors renders correctly 1`] = `
<div
  className="sc-gsnTZi fnwWxp"
>
  <img
    onError={[Function]}
    src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c9/Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg/520px-Bugatti_Veyron_16.4_%E2%80%93_Frontansicht_%281%29%2C_5._April_2012%2C_D%C3%BCsseldorf.jpg"
  />
  <div
    className="sc-dkzDqf cWPqBb"
  >
    <div
      className="sc-bczRLJ hJhdoW"
    >
      brand1
    </div>
    <div
      className="sc-bczRLJ hJhdoW"
    >
      type1
    </div>
  </div>
</div>
`;
